packages/toolpad-studio/src/toolpad/AppEditor/HierarchyExplorer/index.tsx

```diff
@@ -1,22 +1,53 @@
 import * as React from 'react';
+import clsx from 'clsx';
 import { NodeId } from '@toolpad/studio-runtime';
-import { Box, Typography, styled } from '@mui/material';
-import { SimpleTreeView, TreeItem, TreeItemProps } from '@mui/x-tree-view';
+import { Box, Typography, styled, IconButton, SxProps } from '@mui/material';
+import { SimpleTreeView, TreeItem, TreeItemProps, treeItemClasses } from '@mui/x-tree-view';
 import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
 import ChevronRightIcon from '@mui/icons-material/ChevronRight';
 import useBoolean from '@toolpad/utils/hooks/useBoolean';
 import * as appDom from '@toolpad/studio-runtime/appDom';
+import MoreVertIcon from '@mui/icons-material/MoreVert';
+import invariant from 'invariant';
 import { useAppState, useDomApi, useAppStateApi } from '../../AppState';
 import { ComponentIcon } from '../PageEditor/ComponentCatalog/ComponentCatalogItem';
 import { DomView } from '../../../utils/domView';
 import { removePageLayoutNode } from '../pageLayout';
-import EditableTreeItem from '../../../components/EditableTreeItem';
+import EditableTreeItem, { EditableTreeItemProps } from '../../../components/EditableTreeItem';
 import ExplorerHeader from '../ExplorerHeader';
+import NodeMenu from '../NodeMenu';
 
 const CollapseIcon = styled(ExpandMoreIcon)({ fontSize: '0.9rem', opacity: 0.5 });
 const ExpandIcon = styled(ChevronRightIcon)({ fontSize: '0.9rem', opacity: 0.5 });
 
-export interface CustomTreeItemProps extends TreeItemProps {
+const classes = {
+  treeItemMenuButton: 'Toolpad__HierarchyListItem',
+  treeItemMenuOpen: 'Toolpad__HierarchyListItemMenuOpen',
+};
+
+const StyledTreeItem = styled(EditableTreeItem)({
+  [`& .${classes.treeItemMenuButton}`]: {
+    visibility: 'hidden',
+  },
+  [`
+  & .${treeItemClasses.content}:hover .${classes.treeItemMenuButton},
+      & .${classes.treeItemMenuOpen}
+    `]: {
+    visibility: 'visible',
+  },
+});
+
+interface StyledTreeItemProps extends TreeItemProps, EditableTreeItemProps {
+  labelTextSx?: SxProps;
```

Comment:
To simplify this a bit, I think you can just supply values for these directly to the `NodeMenu` component. I see we're exporting `CustomTreeItemProps`, I don't think this is necessary, we can keep the component local to this module for now.